internal/service/s3/bucket_policy.go

```diff
@@ -93,6 +93,12 @@ func resourceBucketPolicyRead(d *schema.ResourceData, meta interface{}) error {
 		Bucket: aws.String(d.Id()),
 	})
 
+	if !d.IsNewResource() && tfresource.NotFound(err) {
```

Comment:
here we'll need to catch the specific AWS error code instead as `tfresource.NotFound` looks for specific errors created with the `tfresource` package e.g.
```suggestion
	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, s3.ErrCodeNoSuchBucket) {
```